### ISSUE TITLE
GNU license is too restrictive

### ISSUE THREAD

**Issue by neaumusic** [wappalyzer/wappalyzer#7875]
I think this library is a great idea, but I realized the weekly downloads are extremely low on npm, and I'm thinking the reason is that the license is too restrictive

I know this is a big ask, but similar to bowser, aria-query, dom-accessibility-api etc, I think it would be great if the license was more permissive!

I wish I noticed the license earlier because I already put in a lot of effort understanding this code, but it's not possible for us to use this package with the current license. It is totally understandable to retain the license, but I figured I'd ask

**Comment by AliasIO** [comment]
I'm not opposed to it but we've had so many [contributors](https://github.com/wappalyzer/wappalyzer/graphs/contributors) over the years that re-licensing will be very difficult. What limitation of GPL are you running in to?

**Comment by TechnologyClassroom** [comment]
The statistics are very good for Wappalyzer. [Chrome](https://chrome.google.com/webstore/detail/wappalyzer-technology-pro/gppongmhjkpfnbhagpmjfkannfbllamg) and [Firefox](https://addons.mozilla.org/en-US/firefox/addon/wappalyzer/) stats show 2,105,346+ users.

See also https://github.com/wappalyzer/wappalyzer/issues/2977

**Comment by neaumusic** [comment]
> I'm not opposed to it but we've had so many [contributors](https://github.com/wappalyzer/wappalyzer/graphs/contributors) over the years that re-licensing will be very difficult. What limitation of GPL are you running in to?

The limitation is that if we use wappalyzer to identify components on a web page, I believe the license dictates that our entire project has to be open source. We have a lot of proprietary and unrelated code, and it doesn't make sense to go open source (unminified/unobfuscated?) so that we can identify components with wappalyzer

**Comment by AliasIO** [comment]
> The limitation is that if we use wappalyzer to identify components on a web page, I believe the license dictates that our entire project has to be open source.

Not at all. It just means that if you distribute copies of Wappalyzer code, including any modifications, the license should be retained and source code provided. You don't have to open-source your own code and if you only use Wappalyzer internally you don't have to do anything.

The idea is that if you take our code and build on it, you contribute back.

**Comment by neaumusic** [comment]
I feel like I understand the intent / what you're saying, but I can't convince myself that's what it means --

Pretty much, distributed software utilizing the code must also be released under GPL, meaning the rest of our code would need to be open source, not just the wappalyzer code. I don't think we can bundle it or include it on the side of our project, unminified (which I think would be fair and serve the purpose you described).

I'm trying to form an argument for why I feel that this GPL license is reaching farther than it should (the question of whether or not this license does actually promote contribution / anti-theft). One example is using the code in a proprietary server context where only responses are sent, and no distributed code (as in a browser extension, electron app, iOS app, etc). How does GPL promote contribution if it requires a lot of overhead sending queries and getting responses from server code?

Anyways, just food for thought. In response to that other comment, I meant the npm package downloads (not the webstore downloads, wappalyzer is a great extension)
